### PR TITLE
Compatibility for updated APIs

### PR DESCRIPTION
**Pull request by emm1R**
Changes in the SD Connect and SD Submit APIs require updates in the filesystem. SD Connect merged its two APIs and requires additional request headers. SD Submit requires that the possible scheme be removed from the dataset name when requesting files.

### CODE REVIEW

**Review by blankdots — COMMENTED** [review]
if i start the UI without `SDS_ACCESS_TOKEN` i get
```
➜ ./cmd/qt/deploy/linux/qt_project
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x6ae1d4]

goroutine 1 [running, locked to thread]:
net/http.(*Client).deadline(0x203000)
	net/http/client.go:189 +0x14
net/http.(*Client).do(0x0, 0xc0004d6500)
	net/http/client.go:611 +0x225
net/http.(*Client).Do(...)
	net/http/client.go:593
net/http.(*Client).Head(0xc00001a1d2, {0xc00001a1d2, 0x9667a0})
	net/http/client.go:934 +0x6a
sda-filesystem/internal/api.glob..func9({0xc00001a1d2, 0x27})
	sda-filesystem/internal/api/api.go:200 +0x2e
sda-filesystem/internal/api.(*sdConnectInfo).getEnvs(0xc000076230)
	sda-filesystem/internal/api/sdconnect.go:105 +0x87
sda-filesystem/internal/api.glob..func6({0xa9eec6, 0xacadaf})
	sda-filesystem/internal/api/api.go:109 +0x42
main.(*LoginModel).checkEnvs(0xc0001012f0)
	sda-filesystem/cmd/qt/loginmodel.go:114 +0x7d
main.main()
	sda-filesystem/cmd/qt/main.go:226 +0x288
```
is this intended ?

**Comment by emm1R** [comment]
Fixed

File: internal/api/sdsubmit_test.go [hunk 1]
```diff
@@ -156,23 +152,50 @@ func Test_SDSubmit_GetFiles_Pass(t *testing.T) {
...
+		t.Fatalf("Function failed, expected no error, received=%v", err)
+	}
+	if len(meta) != 1 {
+		t.Fatalf("Function failed, expected=%d, received=%d", 1, len(meta))
+	}
+	if meta[0].Name != testFile[0].DisplayFileName {
+		t.Errorf("Function failed, expected=%s, received=%s", testFile[0].DisplayFileName, meta[0].Name)
+	}
+	if s.fileIDs["https://dataset1_file1.txt"] != "file1" {
+		t.Errorf("Function failed, expected=%s, received=%s", "file1", s.fileIDs["dataset1_file1.txt"])
```
Comment (teemukataja) [inline internal/api/sdsubmit_test.go:198]:
```suggestion
		t.Errorf("Function failed, expected=%s, received=%s", "file1", s.fileIDs["https://dataset1_file1.txt"])
```
Update error message key name also

**Review by teemukataja — APPROVED** [review]

**Review by blankdots — APPROVED** [review]
All good, might want to address the change that @teemukataja mentioned, other than that, maybe the changelog.

**Pull request merged**